sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -600,6 +600,7 @@ class SparkSession private(
    * @since 2.0.0
    */
   def sql(sqlText: String): DataFrame = {
+    SparkSession.setActiveSession(this)
     val tracker = new QueryPlanningTracker
     val plan = tracker.measurePhase(QueryPlanningTracker.PARSING) {
       sessionState.sqlParser.parsePlan(sqlText)
```

Comment:
I know that we use `SQLConf.get` a lot in analyzer/optimizer rules, but does it also true for the parser?

Reply:
> I know that we use `SQLConf.get` a lot in analyzer/optimizer rules, but does it also true for the parser?

Parser use `SQLConf.get` too.
It truly make some  configuration about parser mess in SparkThriftServer multi thread mode.

Reply:
shall we define a `conf: SQLConf` method in `ParserDriver` so that we can avoid calling `SQLConf.get` in the parser?

Reply:
> shall we define a `conf: SQLConf` method in `ParserDriver` so that we can avoid calling `SQLConf.get` in the parser?

How about new change?  Test is ok for #26172 . also cc @juliuszsompolski